src/jv.c

```diff
@@ -862,14 +862,14 @@ static jv* jvp_array_write(jv* a, int i) {
     int j;
     for (j = 0; j < jvp_array_length(*a); j++) {
       new_array->elements[j] =
-        jv_copy(array->elements[j + jvp_array_offset(*a)]);
+        jv_unborrow(array->elements[j + jvp_array_offset(*a)]);
```

Comment:
Ditto.

Reply:
Need to check that `jvp_refcnt_unshared()` returns false when the `jv` is borrowed...

Reply:
does that always need/ought to be checked? If so it should be added to jv_borrow()

Reply:
No.

Reply:
Actually, this solution was improper to begin with. `jvp_*` should not do borrow checking and unborrowing is unnecessary since the elements of the array will already not be borrowed. I've reverted the change to `jvp_array_write()` and have added the proper check for `jv_array_set()`. I don't really understand what the `jvp_refcnt_unshared()` was all about, but since borrowing is now all handled outside of `jvp_*`  it should not matter.